allauth/headless/mfa/views.py

```diff
@@ -44,14 +45,21 @@ def get(self, request, *args, **kwargs):
         return response.AuthenticatorsResponse(request, authenticators)
 
 
+# TODO: Enforce reauthentication
 class ManageTOTPView(AuthenticatedAPIView):
     input_class = {"POST": ActivateTOTPInput}
 
     def get(self, request, *args, **kwargs):
         authenticator = self._get_authenticator()
         if not authenticator:
+            adapter: DefaultMFAAdapter = get_adapter()
             secret = totp.get_totp_secret(regenerate=True)
-            return response.TOTPNotFoundResponse(request, secret)
+            totp_url: str = adapter.build_totp_url(
+                adapter.get_totp_label(request.user),
+                adapter.get_totp_issuer(),
+                secret,
```

Comment:
I think it would be good if we change the signature of `build_totp_url()` to just accept `user` and `secret`, because inside that method we can just call `self.get_totp_label()` and so on.